src/main/java/org/assertj/core/condition/VerboseCondition.java

```diff
@@ -0,0 +1,158 @@
+      return verbose(expectedValue,matchesBiPredicate,"",null,null);
+  }
+  
+  /**
+   * Creates a new <code>{@link VerboseCondition}</code>
+   *
+   * @param <T> the type of object the given condition accept.
+   * @param <EXPECTED> the type of object the expected value has.
+   * @param expectedValue the object the given value would be tested against.
+   * @param matchesBiPredicate the BiPredicate that tests the value to test against the expected value.
+   * @param matchDescription describes the action of the matchesBiPredicate verbal.
+   * @return the created {@code VerboseCondition}.
+   * @throws NullPointerException if the matchesBiPredicate is {@code null}.
+   */
+  public static <T,EXPECTED> VerboseCondition<T,EXPECTED> verbose(EXPECTED expectedValue, BiPredicate<T, EXPECTED> matchesBiPredicate,
+          String matchDescription) {
+      return verbose(expectedValue,matchesBiPredicate,matchDescription,null,null);
+  }
+  
+  /**
+   * Creates a new <code>{@link VerboseCondition}</code>
+   *
+   * @param <T> the type of object the given condition accept.
+   * @param <EXPECTED> the type of object the expected value has.
+   * @param expectedValue the object the given value would be tested against.
+   * @param matchesBiPredicate the BiPredicate that tests the value to test against the expected value.
+   * @param matchDescription describes the action of the matchesBiPredicate verbal.
+   * @param expectedValueTransformation Function that transforms the expected value into a text.
+   * @param actualValueTransformation  Function that transforms the value to test into a text.
+   * @return the created {@code VerboseCondition}.
+   * @throws NullPointerException if the matchesBiPredicate is {@code null}.
+   */
+  public static <T,EXPECTED> VerboseCondition<T,EXPECTED> verbose(EXPECTED expectedValue, BiPredicate<T, EXPECTED> matchesBiPredicate,
+          String matchDescription, Function<EXPECTED, ?> expectedValueTransformation,
+          Function<T, ?> actualValueTransformation) {
+      return new VerboseCondition<T, EXPECTED>(expectedValue,matchesBiPredicate,matchDescription,expectedValueTransformation,actualValueTransformation);
+  }
+  
+  private VerboseCondition(EXPECTED expectedValue, BiPredicate<T, EXPECTED> matchesBiPredicate,
+      String matchDescription, Function<EXPECTED, ?> expectedValueTransformation,
+      Function<T, ?> actualValueTransformation) {
+    requireNonNull(matchesBiPredicate, "The matchesBiPredicate should not be null");
+    this.expectedValue = expectedValue;
+    this.matchBiPredicate = matchesBiPredicate;
+    this.matchDescription = matchDescription;
+    this.expectedValueTransformation = expectedValueTransformation;
+    this.actualValueTransformation = actualValueTransformation;
+    describedAs("%s <%s>",matchDescription,
+        transformIf(expectedValueTransformation, expectedValue));
```

Comment:
do you need this as it is going to be overridden when matches is called?

Reply:
If i am using an (non-soft) `AllOf(c1,c2,c3)` and c1 fails a condition would be not executed. to it would be nice to have a description.